astro/terraform/terraform_remote_state_disable_utils.go

```diff
@@ -66,24 +69,51 @@ func deleteTerraformBackendConfig(in []byte) (updatedConfig []byte, err error) {
 		return nil, errors.New("could not parse \"terraform\" block in config")
 	}
 
-	if err := astDel(terraformConfigBlock.List, "backend"); err != nil {
-		return nil, err
-	}
+	astDelIfExists(terraformConfigBlock.List, "backend")
 
 	buf := &bytes.Buffer{}
 	printer.Fprint(buf, config)
 
 	return buf.Bytes(), nil
 }
 
-func deleteTerraformBackendConfigFromFile(file string) error {
+// hcl2 (used by terraform 0.12) doesn't provide interface to walk through the AST or
+// to modify block values, see https://github.com/hashicorp/hcl2/issues/23 and
+// https://github.com/hashicorp/hcl2/issues/88
+// As a work around we'll perform surgery directly on text, if backend config is simple.
+// The method returns an error, if the config is too complicated to be parsed with the regexp.
+// This method should be rewritten once hcl2 supports AST traversal and modification.
+func deleteTerraformBackendConfigWithHCL2(in []byte) (updatedConfig []byte, err error) {
+	// Regexp to find if any backend configuration exists
+	backendDefinitionRe := regexp.MustCompile(`(?s)[{\s+]backend\s+"[^"]+"\s*{`)
+	// Regexp to find simple backend configuration, which doesn't contain '{}' inside
+	backendBlockRe := regexp.MustCompile(`(?s)(\s*backend\s+"[^"]+"\s*{[^{]*?})`)
```

Comment:
would it be easier to, instead of regex, iterate the input as follows:

```
ix = in.find("\"backend\"")
bracket_count = 0
for ; ix<len(in); ix++ {
  if in[ix]=='{' {
    bracket_count++
  }
  if in[ix]=='}' {
    bracket_count--
    if bracket_count==0 {
      end_ix = ix
      break
    }
  }
}
```

That way we support complicated configs as well, right? There are some edge cases of course, but this covers a lot more expected cases.

This does assume that the config is correct HCL of course.

Reply:
Yep, I thought about this solution. It actually doesn't look as bad at I thought it would.
It's still not ideal, because '{' or '}' can be part of some other lexeme. But in reality this is very unlikely. I'll change the code to what you've suggested, thanks!

Reply:
I implemented suggested approach and got to writing tests, and I'm not sure if it's a good approach anymore.
The problem is that this config
```
terraform {
         backend "local" {
           constant = "weird_string_{"
         }
       }
```
Will be recognized as correct expression and turned into
```
terraform {
```
I think it's better to fail explicitly in this case.
So once the HCL2 library is ready we should parse config using it, but until then, I'd stick with regexp on simple configs and failures on complex ones.